server/controllers/controllers.js

```diff
@@ -18,12 +18,43 @@ exports.getTopics = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  fetchArticles()
-    .then((articles) => {
-      res.status(200).send({ articles });
+    if (req.query.hasOwnProperty("sort_by")) {
+        
+        const validSortBy = ["article_id", "title", "topic", "author", "body", "created_at", "votes", "article_img_url"];
+
+            if (!validSortBy.find(column => column === req.query.sort_by)) {
+                next("That column doesn't exist");
+            };
+    };
+
+    if (req.query.hasOwnProperty("order")) {
+        
+        const validOrder = ["ASC", "DESC"];
+
+            if (!validOrder.find(option => option === req.query.order.toUpperCase())) {
+                next("Invalid order");
+            };
+    };
+
+    queries = req.query; 
+    
+    const topicsCheck = fetchTopics()
```

Comment:
This logic is okay but it does mean some error handling in the controller. A check exists function would be good here which would reject if the topic doesn't exist. Worth considering a refactor at a later date.